.github/workflows/ci.yml

```diff
@@ -1,15 +1,34 @@
-name: ci
+name: CI Workflow
 
 on:
   pull_request:
     branches:
       - main
 
-  workflow_dispatch:
-
 jobs:
+  block-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if PR is from forked repo
+        id: check_forked
+        run: |
+          if [[ ${{ github.event.pull_request.head.repo.full_name }} != ${{ github.event.pull_request.base.repo.full_name }} ]]; then
+            echo "{is_forked}=={true}" >> $GITHUB_OUTPUT
+          else
+            echo "{is_forked}=={false}" >> $GITHUB_OUTPUT
+          fi
+      - name: Block PR from forked repo
+        if: steps.check_forked.is_forked == 'true'
+        run: |
+          echo "This pull request was pushed from a forked repository, which is not allowed. Please push changes from a branch in the upstream repository"
+          exit 1
+      - name: Pass PR from upstream repo
```

Comment:
And a third step. This one could potentially go into the first one as well.
Less steps, less code, less time to run. WDYT?